File: 2021-CCC/j2-2021.py
def solve(names,bids):
    highest = 0
    Name = ""
    for i in range(len(names)):
        if bids[i]>highest:
            x = bids[i]
            highest = 0
            highest += x
            Name = names[i]
    return Name

names = []
bids = []
people = int(input())

for i in range(people):
    name = input()
    bid = int(input())
    names.append(name)
    bids.append(bid)
print(solve(names,bids))

File: 2021-CCC/j2-2021.cpp
#include <iostream>
#include <map>
#include <string>
using namespace std;


string solve(int prices[],string names[],int people){
    int cost = 0;
    string name = "";
    for(int x=0;x<people;x++){
        if (prices[x]>cost){
            cost = prices[x];
            name = names[x];
        }
    }
    return name;
}


int main(){
    int people;
    cin >> people;
    string names[people];
    int prices[people];
    string name;
    int price;
    for(int i=0;i<people;i++){
        cin >> name;
        cin >> price;
        names[i] = name;
        prices[i] = price;
    }
    cout << solve(prices,names,people) << endl;



}